### PR TITLE
Bottomup nms

### PR DESCRIPTION
**Pull request by luminxu**
add nms for bottom up dataset and demo.

### CODE REVIEW

**Review by jin-s13 — APPROVED** [review]

File: mmpose/apis/inference.py [hunk 1]
```diff
@@ -492,11 +495,19 @@ def inference_bottom_up_pose_model(model,
...
+            area = (np.max(pred[:, 0]) - np.min(pred[:, 0])) * (
+                np.max(pred[:, 1]) - np.min(pred[:, 1]))
             pose_results.append({
                 'keypoints': pred[:, :3],
+                'score': result['scores'][idx],
+                'area': area,
             })
 
+        # pose nms
+        keep = oks_nms(list(pose_results), oks_thr, sigmas=None)
```
Comment (ly015) [inline mmpose/apis/inference.py:508]: pose_results should be list already?

Comment (jin-s13) [reply mmpose/apis/inference.py:508]: It seems that `list` is not necessary here.

Comment (luminxu) [reply mmpose/apis/inference.py]: I have deleted `list` in inference and (bottom up & top down) dataset files.

**Review by ly015 — COMMENTED** [review]
Ok the image-level NMS can not be applied in a top-down detector for now. So it's okay to do it in the dataset.

**Review by ly015 — APPROVED** [review]

File: demo/bottom_up_img_demo.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def main():
     parser.add_argument(
         '--kpt-thr', type=float, default=0.3, help='Keypoint score threshold')
     parser.add_argument(
-        '--oks-thr', type=float, default=0.9, help='NMS oks threshold')
+        '--pose-nms-thr', type=float, default=0.9, help='Pose nms threshold')
```
Comment (ly015) [inline demo/bottom_up_img_demo.py:37]: I suggest we make it clear that pose-nms-thr is applied on OKS score in the help message.

File: demo/bottom_up_pose_tracking_demo.py [hunk 3]
```diff
@@ -28,26 +28,14 @@ def main():
     parser.add_argument(
         '--kpt-thr', type=float, default=0.5, help='Keypoint score threshold')
     parser.add_argument(
-        '--nms-oks-thr', type=float, default=0.9, help='NMS oks threshold')
+        '--pose-nms-thr', type=float, default=0.9, help='Pose nms threshold')
```
Comment (ly015) [inline demo/bottom_up_pose_tracking_demo.py:31]: I suggest we make it clear that pose-nms-thr is applied on OKS score in the help message.

**Pull request merged**